iothub_client/inc/iothub_client_core_common.h

```diff
@@ -17,137 +17,138 @@ extern "C"
 {
 #endif
 
-#define IOTHUB_CLIENT_FILE_UPLOAD_RESULT_VALUES \
-    FILE_UPLOAD_OK, \
-    FILE_UPLOAD_ERROR
+typedef enum IOTHUB_CLIENT_FILE_UPLOAD_RESULT_TAG
+{
+    FILE_UPLOAD_OK = 0,
+    FILE_UPLOAD_ERROR = 1
+} IOTHUB_CLIENT_FILE_UPLOAD_RESULT;
 
-    MU_DEFINE_ENUM_WITHOUT_INVALID(IOTHUB_CLIENT_FILE_UPLOAD_RESULT, IOTHUB_CLIENT_FILE_UPLOAD_RESULT_VALUES)
-        typedef void(*IOTHUB_CLIENT_FILE_UPLOAD_CALLBACK)(IOTHUB_CLIENT_FILE_UPLOAD_RESULT result, void* userContextCallback);
+const char* IOTHUB_CLIENT_FILE_UPLOAD_RESULTStrings(IOTHUB_CLIENT_FILE_UPLOAD_RESULT value);
```

Comment:
so these functions just become NOP's?

Reply:
what are those ENUMString()?

Reply:
This is what MU_ENUM_TO_STRINGS() ends up calling once the macro unexpanded.  By putting these in, it means that existing code will just work.

---
In reply to: [350328454](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r350328454) [](ancestors = 350328454)

Reply:
Yes.  MU_ENUM_TO_STRING will just work - 0 changes - with these.

---
In reply to: [352722212](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r352722212) [](ancestors = 352722212)

Reply:
Kicking off offline discussion about best way to handle this in terms of comments, should it be at bottom of file, etc etc.?

---
In reply to: [353488125](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r353488125) [](ancestors = 353488125,352722212)